accepted/0000-hub_retail.md

```diff
@@ -0,0 +1,213 @@
+
+Use cases currently not covered:
+
+- Move image tarball to external HTTP server
+  - In large environments it makes sense to use one or more servers for distribution of images, it should not be the Hub itself.
+- Copy image metadata from one SUMA Server to another 
+  - Hub - copy to all servers
+  - Test in lab, then move the same image to production.
+- Adjust deployment details
+
+
+Proposed solution:
+
+- Implement API calls to
+  - Get Kiwi Image pillar
+  - Update Kiwi Image pillar
+  - Export Kiwi Image (complete metadata)
+  - Import Kiwi Image
+
+- Implement a tool that handles the use cases above
+
+
+HWTYPE configurations
+---------------------
+
+This is currently implemented as a formula assigned to a group with special name.
+
+FIXME: pick one alternative
+
+Alternative 1:
+Extend retail_yaml to create the groups and formula data via Hub and Peripheral Server API.
+This is relatively small change to the code.
+
+Alternative 2:
+Central creation of groups is covered in [Hub Users RFC](00061-hub-users.md). The salt states
+can be extended to handle also formula data.
+This approach is probably more flexible.
+
+Both alternatives will have to cope with authentization and Hub vs. Peripheral server user credentials.
+
+YAML configuration
+------------------
+
+FIXME: optional, related to Alternative 2 in previous section
+
+We have retail_yaml tool for initial configuration of Retail environment from YAML file.
+[Hub Users RFC](00061-hub-users.md) introduced similar concept, based on Salt.
+Maybe these parts could be merged.
+
+
+Combined SUMA and Branch server
+-------------------------------
+
+Branch services are configured by Salt formulas. Also PXE entries for each terminal are generated
+by Salt (currently via reactor).
+With combined SUMA and Branch server, these formulas should be applied on SUMA server itself.
+
+Terminals connect to SUMA directly, SUMA Proxy is not needed.
+
+
+FIXME: pick one alternative
+
+Alternative 1: Hub as a Salt Master for Combined SUMA and Branch server
+
+- this is already used for [Hub Users RFC](00061-hub-users.md)
+
+problems:
+- does not work for PXE entries and disconnected operation
+- mixed environment with Hub, Peripheral Servers with standalone Branch Serveres and 
+  Combined Peripheral Servers will have non-optimal configuration
+- does not cover the "Single server" setup
+
+
+Alternative 2: Multi-master setup of Combined SUMA and Branch server
+
+- Salt minion running on Combined SUMA and Branch server has 2 masters - Hub and itself
+
+problems:
+- SUMA registered to itself cause many problems
+  - we already have clients that do not allow certain actions (containers) so this should be fixable
+- Salt Multi-master might not be reliable
+
+
+Alternative 3: Use salt-call --local for Branch server part of combined server
+
+- how to represent such setup in GUI and API ?
+- would need significant changes in SUMA code
```

Comment:
Are the changes only about allowing to apply formulas (optionally with forms) via `--local` or do you see other changes? Which?

Reply:
I think it would need changes in code to apply formulas and supply pillars for them, maybe also in handling of salt events.